server/src/main/java/org/apache/druid/server/coordination/SegmentChangeStatus.java

```diff
@@ -71,6 +75,25 @@ public String getFailureCause()
     return failureCause;
   }
 
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    SegmentChangeStatus that = (SegmentChangeStatus) o;
+    return state == that.state && Objects.equals(failureCause, that.failureCause);
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Objects.hash(state, failureCause);
+  }
```

Comment:
I would generally prefer to add `equals`/`hashCode` when it is needed for the actual implementation or if the class has several fields and it makes testing easier.

Since we currently have only 2 fields in this class, we can probably skip it for the time being. Especially since there is only one test which is using this.